x/psiphon/psiphon.go

```diff
@@ -208,21 +88,51 @@ func (d *Dialer) DialStream(unusedContext context.Context, addr string) (transpo
 	return streamConn{netConn}, nil
 }
 
-// Start configures and runs the Dialer. It must be called before you can use the Dialer. It returns when the tunnel is ready.
-func (d *Dialer) Start(ctx context.Context, config *Config) error {
-	configJSON, err := json.Marshal(config)
-	if err != nil {
-		return fmt.Errorf("failed to convert config to JSON")
+func newPsiphonConfig(config *DialerConfig) (*psi.Config, error) {
+	if config == nil {
+		return nil, errors.New("config must not be nil")
 	}
-	pConfig, err := psi.LoadConfig(configJSON)
+	// Validate keys. We parse as a map first because we need to check for the existence
+	// of certain keys.
+	var configMap map[string]interface{}
+	if err := json.Unmarshal(config.ProviderConfig, &configMap); err != nil {
+		return nil, fmt.Errorf("failed to parse config: %w", err)
+	}
+	for key, value := range configMap {
+		switch key {
+		case "DisableLocalHTTPProxy", "DisableLocalSocksProxy":
```

Comment:
We are still digging into a specific field in the config. Not sure whether this is breaking the "black-box" or not.

Reply:
Because those fields are incompatible with the Outline SDK. The key change in this PR is from specifying what fields to allow to specifying what fields to now allow. So Psiphon config - disallowed fields is a black box now.